### PR TITLE
Crowdin incontext

### PR DESCRIPTION
**Pull request by annaxuphoto**
PR for FND hack day project: enable Crowdin In-Context tool on dashboard pages. Users will be able to choose the pseudo-locale "in-TL" to enable the In-Context tool, which shows the original & translated strings in Crowdin for a given language.

Example Courses page:
Screen Shot 2020-12-14 at 12 04 56 PM

Note that this is an incomplete implementation of the In-Context tool, hence the absence of page content.

## Links

- [spec]()
- [jira]()

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/config/locales.yml [hunk 1]
```diff
@@ -191,6 +191,11 @@
   english: "Indonesian"
   native: "Bahasa Indonesia"
 
+"in": "in-TL"
+"in-TL":
+  english: "Crowdin In-Context"
+  native: "Crowdin In-Context"
+
```
Comment (daynew) [inline dashboard/config/locales.yml:199]: This should probably have the "debug" attribute added to it so it doesn't show up for normal students. See the other configurations as an example.

File: bin/i18n/hoc_sync_utils.rb [hunk 2]
```diff
@@ -16,7 +16,7 @@ def self.sync_in
     dest_dir = File.join(I18N_SOURCE_DIR, "hourofcode")
 
     # Copy the file containing developer-added strings
-    Dir.mkdir(dest_dir) unless Dir.exist?(dest_dir)
+    FileUtils.mkdir_p(dest_dir)
```
Comment (daynew) [inline bin/i18n/hoc_sync_utils.rb]: Sorry could you remove this? It is just a general sync script fix. I will do that in a separate PR.

File: bin/i18n/sync-in.rb [hunk 3]
```diff
@@ -223,6 +223,7 @@ def localize_level_content
     end
   end
 
+  FileUtils.mkdir_p(File.join(I18N_SOURCE_DIR, "dashboard"))
```
Comment (daynew) [inline bin/i18n/sync-in.rb]: Same comment for this file

File: dashboard/app/views/layouts/application.html.haml [hunk 4]
```diff
@@ -1,6 +1,11 @@
 !!! 5
 %html{dir: locale_dir}
   %head
+    - if request.locale === "in-TL"
+      :javascript
+        var _jipt = [];
+        _jipt.push(['project', 'codeorg']);
+      %script{type: "text/javascript", src: "//cdn.crowdin.com/jipt/jipt.js"}
```
Comment (daynew) [inline dashboard/app/views/layouts/application.html.haml:10]: Could you add a quick comment here which links to https://crowdin.com/project/codeorg/settings#in-context so future developers kind learn more about what we are doing here? Maybe also add a comment that `in-TL` is a pseudo-language locale used for supporting the Crowdin In-Context translation tool?

**Review by daynew — APPROVED** [review]
Looks good! Please address my comments before merging.

File: dashboard/app/views/layouts/application.html.haml [hunk 5]
```diff
@@ -1,6 +1,13 @@
 !!! 5
 %html{dir: locale_dir}
   %head
+    // The Crowdin In-Context tool (https://crowdin.com/project/codeorg/settings#in-context) allows us to view original & translated strings in Crowdin.
+    // "in-TL" is a pseudo-language locale where the Crowdin strings are stored
+    - if request.locale === "in-TL"
+      :javascript
+        var _jipt = [];
```
Comment (hacodeorg) [inline dashboard/app/views/layouts/application.html.haml:8]: Just for my curiosity, what does `jipt` stand for?

Comment (annaxuphoto) [reply dashboard/app/views/layouts/application.html.haml:8]: Just In Place Translations

**Pull request merged**